src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -0,0 +1,194 @@
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Cross references the user's current list of Modules against the given
+     * modules argument and marks out any undone Modules. Displays a successMessage
+     * if all modules are done, and a failMessage if there are left over modules.
+     * @param modules List of Modules of a certain category (Eg. Foundation, IT Professsionalism)
+     * @param failMessage Fail message for particular category of Modules.
+     * @param successMessage Success message for particular category of Modules.
+     */
+    public void compareModules(ObservableList<Module> modules, String failMessage, String successMessage) {
+        boolean areModulesCleared = true;
+        StringBuilder modulesToAdd = new StringBuilder();
+        for (Module module : modules) {
+            if (!doesModuleAlreadyExist(module, currentModules)) {
+                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+                modulesToAdd.append("\n").append(moduleToAdd);
+                areModulesCleared = false;
+            }
+        } if (areModulesCleared) {
+            leftOverModules += successMessage + "\n";
+        } else {
+            leftOverModules += failMessage + modulesToAdd + "\n";
+        }
+        leftOverModules += "\n";
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Science Modules.
+     */
+    public void compareScience(ObservableList<Module> requiredScience) {
+        boolean areModulesCleared = true;
+        for (Module module : requiredScience) {
+            if (doesModuleAlreadyExist(module, currentModules)) {
+                leftOverModules += MESSAGE_SUCCESS_SCIENCE + "\n";
+                areModulesCleared = false;
+                break;
+            }
+        } if (areModulesCleared) {
+            leftOverModules += MESSAGE_SCIENCE + "\n";
+        }
+        leftOverModules += "\n";
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Internship Modules. Also calculates current MC score
+     * achieved from Internship Modules.
+     */
+    public void compareInternship(ObservableList<Module> requiredInternship) {
+        int modularScore = 0;
+        StringBuilder leftOverInternship = new StringBuilder();
+        for (Module module : requiredInternship) {
+            boolean areModulesCleared = true;
+            if (doesModuleAlreadyExist(module, currentModules)) {
+                int modularCredits = Integer.parseInt(module.getModularCredits().toString());
+                modularScore += modularCredits;
+                areModulesCleared = false;
+            }
+            if (areModulesCleared) {
+                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+                leftOverInternship.append("\n").append(moduleToAdd);
+            }
```

Comment:
Actually how come we need the `areModulesCleared`flag instead of using an if-else statement ahhh? 

Just to clarify, is this code saying: if a module is taken, then add its MCs to `modularScore`. If it's not, add the module title/mc to the display instead.

PAISEH I also very confused here ><

```suggestion
            if (doesModuleAlreadyExist(module, currentModules)) {
                int modularCredits = Integer.parseInt(module.getModularCredits().toString());
                modularScore += modularCredits;
            } else { 
                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
                leftOverInternship.append("\n").append(moduleToAdd);
            }
```